# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/frame_builder.rs
# Commit: e24ed89246a4
# Full Hash: e24ed89246a4a6c6b4a567ec412f0e3295b4570d
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-04-26 09:46:09
# Regressor Bug: 1766017
# File Overlap Count: 1
# Description:
#   Bug 1766017 - Do a better job of clipping off-screen surface allocation sizes r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D144534
# ==============================================================================

diff -r cdc42b7822f1 -r e24ed89246a4 gfx/wr/webrender/src/frame_builder.rs
--- a/gfx/wr/webrender/src/frame_builder.rs	Mon Apr 25 22:28:35 2022 +0000
+++ b/gfx/wr/webrender/src/frame_builder.rs	Mon Apr 25 23:40:40 2022 +0000
@@ -337,7 +337,7 @@
                         // re-map the dependencies of any child primitives.
                         let surface = &scene.surfaces[surface_index.0];
                         let world_culling_rect = tile_cache.pre_update(
-                            surface.local_rect,
+                            surface.unclipped_local_rect,
                             surface_index,
                             &visibility_context,
                             &mut visibility_state,